src/Nethermind/Nethermind.Pipeline/Publishers/WebSocketsPublisher.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Nethermind.PubSub;
+using Nethermind.Serialization.Json;
+using Nethermind.WebSockets;
+
+namespace Nethermind.Pipeline.Publishers
+{
+    public class WebSocketsPublisher<T> : IPipelineElement<T> ,IPublisher, IWebSocketsModule
```

Comment:
I think that you can have IPipelineSource<TIn>, IPipeline<TIn, TOut>, IPipelinePublisher<TIn>